### PR TITLE
Fix: Android changed canvas to non-optional

### PR DESCRIPTION
**Pull request by braandl**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -26,6 +26,11 @@ def getExtOrIntegerDefault(name) {
 }
 
 android {
+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+      namespace "com.reactnativeshadowview"
+  }
```
Comment (dimaportenko) [inline android/build.gradle:32]: what is this code for?

File: android/build.gradle [hunk 2]
```diff
@@ -127,5 +132,5 @@ dependencies {
   // noinspection GradleDynamicVersion
   api 'com.facebook.react:react-native:+'
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-  implementation "androidx.core:core-ktx:$kotlin_version"
+  implementation "androidx.core:core-ktx:1.7.0"
```
Comment (dimaportenko) [inline android/build.gradle]: @braandl why is it good idea to move from a variable to hardcoded version?

File: android/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,4 +1,2 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="com.reactnativeshadowview">
-
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (dimaportenko) [inline android/src/main/AndroidManifest.xml:1]: why do we need this change?

File: android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt [hunk 4]
```diff
@@ -260,7 +260,7 @@ open class ShadowLayout @JvmOverloads constructor(
             shadow_y_shift / shadow_downscale / cssRatio
         )
 
-    override fun draw(canvas: Canvas?) {
+    override fun draw(canvas: Canvas) {
```
Comment (dimaportenko) [inline android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt:263]: I don't mind merging this PR, but would like to hear what those changes solves.

Comment (braandl) [reply android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt:263]: The Canvas Element is no more optional when compiling against latest android SDK (34) and therefore the build fails. That is why it was changed here.

Comment (dimaportenko) [reply android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt:263]: @braandl got it. Are other changes redundant?

Comment (braandl) [reply android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt:263]: You can remove the other changes from the PR if you wish. They are mainly auto optimizations done by AS. I can also do it for you if you prefer.

File: android/src/main/AndroidManifest.xml (hunk 3 above)
Comment (braandl) [reply android/src/main/AndroidManifest.xml:1]: This is also not needed. It was removed by AS.

File: android/build.gradle (hunk 2 above)
Comment (braandl) [reply android/build.gradle]: It's not. I think this slipped through, I am sorry.

File: android/build.gradle (hunk 1 above)
Comment (braandl) [reply android/build.gradle:32]: Just checking the Android Grade Plugin version to only define the namespace when it is supported by AGP, so you can build with AGP Version 5.X

File: android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt (hunk 4 above)
Comment (dimaportenko) [reply android/src/main/java/com/reactnativeshadowview/ShadowLayout.kt:263]: yeah, would be nice if you can clean up PR and I'll try to test and merge it on the weekend

**Comment by dimaportenko** [comment]
@braandl I did merge and release, but unfortunately can't properly test. Would be great if you can check this works as you expected.

**Comment by braandl** [comment]
Thank you! Works like a charm :+1: 

There are a few more deprecations in newer android versions, I will provide PR asap.

**Pull request merged**